src/lib/components/form/select/select-field.svelte

```diff
@@ -16,25 +22,49 @@
   export let helper = "";
   export let required = false;
   export let isMultiple = false;
+  export let withAutoComplete = false;
+  export let withClearButton = false;
   export let hideLabel = false;
   export let showIconForSelectedOption = false;
   export let choices: Choice[];
+  export let optGroups: Choice[] = [];
   export let display: "horizontal" | "vertical" = "horizontal";
-  export let style: "common" | "filter" = "common";
+  export let style: "common" | "filter" | "search" = "common";
   export let errorMessages: string[] | undefined = undefined;
   export let handleEltChange: (d: { detail: string }) => void | undefined =
     undefined;
 
-  let ariaDescribedBy = "";
+  const originalChoices: Choice[] = [...choices];
+  const originalOptGroups: Choice[] = [...optGroups];
+
+  let optGroupsOpen: string[] = [];
+  let valuesSelectedInOptGroup: Record<string, number> = {};
```

Comment:
```suggestion
  let numValuesSelectedInOptGroup: Record<string, number> = {};
```

Mais je pense que je laisserais tomber tout ce mecanisme de comptage de compliqué, qui au final sert juste à renseigner le  `@const hasSubValuesSelected` l. 340.
Tu pourrais le remplacer par un appel de fonction qui va compter à ce moment là les besoins selectionnés du groupe en cours.
De façon très approximative: 
```
{@const hasSubValuesSelected = optGroupHasSubValuesSelected(optGroup)}
```
avec
```
function countSubValuesSelected(optGroup) {
    return values.filter(v => v.split('--')[0] === optGroup.value).length > 0
}
```

Reply:
En effet, c'est beaucoup plus simple 😅